src/app/static/src/app/store/metadata/metadata.ts

```diff
@@ -38,22 +38,32 @@ export const metadataGetters = {
             return [];
         }
 
-        const selectedDataType = rootState.surveyAndProgram.selectedDataType;
+        const sap = rootState.surveyAndProgram;
+        const selectedDataType = sap.selectedDataType;
 
         let metadataForType: Metadata | null = null;
+        let dataIndicators: FilterOption[];
         switch (selectedDataType) {
             case (DataType.ANC):
                 metadataForType = plottingMetadata.anc;
+                dataIndicators = sap.anc?.filters.indicators || [];
                 break;
             case (DataType.Program):
                 metadataForType = plottingMetadata.programme;
+                dataIndicators = sap.program?.filters.indicators || [];
                 break;
             case (DataType.Survey):
                 metadataForType = plottingMetadata.survey;
+                dataIndicators = sap.survey?.filters.indicators ||[];
                 break;
         }
 
-        return (metadataForType && metadataForType.choropleth) ? metadataForType.choropleth.indicators : [];
+        const unfiltered =  (metadataForType && metadataForType.choropleth) ? metadataForType.choropleth.indicators : [];
```

Comment:
Is the generated `Metadata` type wrong? It is:
```
export interface Metadata {
  choropleth?: {
    indicators?: ChoroplethMetadata;
    [k: string]: any;
  };
  [k: string]: any;
}
```
which would imply that `unfiltered` here would be of type singular `ChoroplethIndicatorMetadata`, rather than an array. And that `choropleth.indicators` could be undefined. If this is wrong then looks we need to update this hintr schema and regenerate types: https://github.com/mrc-ide/hintr/blob/master/inst/schema/PlottingMetadataResponse.schema.json Also suggests `hintr` is missing a test that validates the plotting metadata against the schema...

Reply:
Yeah, the types are  complete mess. There are two `ChoroplethMetadata`s, which confuses the issue - one a type and one an interface. What we actually seem to get in `indicators` corresponds to the type definition, which is indeed an array of indicators.
I'm not really sure how this is happened, but as you say it needs to be fixed in hintr and types regenerated - I'll open a ticket for that

Reply:
Just looking at the schema again and it looks correct actually, I didn't read it carefully enough. But there's both an internal definition called "choropleth_metadata" and a separate file schema called ChoroplethMetadata, which I guess is causing the duplicate and the weird behaviour. Let me see if updating the hintr schema on a branch and regenerating fixes it, as then we can rely on those typings here and keep things a bit clearer.

Reply:
Yeah, I was just getting there myself. Is the schema obsolete then, can that go to avoid the duplicate? (which causes problems in Typescript)

Reply:
I've added a ticket to fix in hintr, though text might not be accurate now: https://mrc-ide.myjetbrains.com/youtrack/issue/mrc-2787